### PR TITLE
incomplete

### PR DESCRIPTION
**Pull request by MattWellie**
This is currently incomplete, I'll come back to the more fiddly JSON -> TOML transition later

@lgruen could you please confirm if the Docker Secret Credentials currently being used to authenticate GCloud have read & write permissions on `cpg-config`? @illusional following Leo's message yesterday are you able to take this over?

The short term plan I have here is:

1. Convert the Dockers.json files (Azure and GCP) into our CPG config toml format, and post a version to `cpg-config` (e.g. gs://cpg-config/templates/images/gatk-sv/gcp_dockers.toml)
2. Alter the docker building workflow from reading, updating, and re-committing a local JSON file, to reading the current state from the cpg-config bucket, updating as required, then pushing a new version of the TOML.
3. Incorporate a bump2version version file and configuration here to track versioning instead of parsing github releases from the Broad's repository. Instead of making a chunky image tag with several different elements, are we happy to use this bump2 controlled tag as the only image tag?
4. Update all uses of JSON -> TOML in the docker builder script

Once that all works, run the docker builder workflow targeting `all` images - The workflow determines which images to build (& push) based on code changes, but as all the upstream merges happened without building any images we either need to copy current images from the Broad, or build all of our own. 

After all images are built and stored without our infrastructure, revert the workflow to finding appropriate images by commit.

Azure... ? I guess we also need to duplicate these images into Azure, which will require extra auth credentials and the location of the azure artifactory 🙃

### CODE REVIEW

File: inputs/values/azure_dockers.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[images]
```
Comment (illusional) [inline inputs/values/azure_dockers.toml]: My pet peeve against the word `dockers`, can we call it `containers`?
Do we need this as a base? Or can the original base just be an empty file so the first build pushes both?

File: scripts/docker/build_docker.py [hunk 2]
```diff
@@ -72,10 +72,11 @@ class ImageDependencies:
                            that this dockerfile derives from, and the values indicate corresponding ARG values that
                            must be passed to docker_build to specify the correct repo/image:tag
     """
+
     def __init__(
-            self,
-            git_dependencies: Union[str, Iterable[str]],
-            docker_dependencies: Mapping[str, str] = MappingProxyType({})
+        self,
```
Comment (illusional) [inline scripts/docker/build_docker.py]: Should we suggest using black upstream to mitigate conflicts in the future?

**Review by illusional — COMMENTED** [review]
On board with these changes, you'll also need to add it into the analysis-runner (https://github.com/populationgenomics/analysis-runner/blob/3ea52ed77af40d9d9be27a000d3c029fcdb108b3/server/util.py#L249).

File: inputs/values/azure_dockers.toml (hunk 1 above)
Comment (MattWellie) [reply inputs/values/azure_dockers.toml]: Yeah, I hate `dockers` a lot... `images.toml` is the naming convention we're already using, so probably `cpg-config/templates/images/gatk-sv/{gcp/azure}_images.toml`?

Comment (MattWellie) [reply inputs/values/azure_dockers.toml]: Empty file is an interesting one... I'll ping this workflow with a couple of dry runs to see what happens if the 'dockers' file is empty, or if the file has all the correct keys but None for 'current image' values. I think I understand the behaviour, but I'm keen to confirm

File: scripts/docker/build_docker.py (hunk 2 above)
Comment (MattWellie) [reply scripts/docker/build_docker.py]: I'll raise this on their upstream repo

**Pull request closed without merging**